lib/diagnostic.rb

```diff
@@ -116,19 +140,23 @@ def initialize
   end
 
   def punch
-    @moves << 'punch'
-    @damage += 5
-    @damage *= multiplier
+    self.moves << 'punch'
+    self.damage += 5
+    self.damage *= multiplier
   end
 
   def move(direction)
-    @moves << "move #{direction}"
+    self.moves << "move #{direction}"
   end
 
   def kick
-    @moves << 'kick'
-    @damage += 10
-    @damage *= multiplier
+    self.moves << 'kick'
+    self.damage += 10
+    self.damage *= multiplier
+  end
+
+  def get_possible_moves
+    p "kick, move, punch"
```

Comment:
The way to make the method chaining work is for `punch`, `kick`, and `move` to return `self`.That way, `ComboAttack.new.punch.move('left').kick.damage` is equivalent to

```ruby
combo = ComboAttack.new
combo.punch
combo.move('left')
combo.kick
combo.damage
```